crates/web5/src/rfc3339.rs

```diff
@@ -0,0 +1,55 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Deserializer, Serializer};
+use std::time::SystemTime;
+
+pub(crate) fn serialize_system_time<S>(
+    time: &SystemTime,
+    serializer: S,
+) -> std::result::Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let datetime: chrono::DateTime<Utc> = (*time).into();
+    let s = datetime.to_rfc3339();
+    serializer.serialize_str(&s)
+}
+
+pub(crate) fn deserialize_system_time<'de, D>(
+    deserializer: D,
+) -> std::result::Result<SystemTime, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    let datetime = chrono::DateTime::parse_from_rfc3339(&s).map_err(serde::de::Error::custom)?;
+    Ok(datetime.with_timezone(&Utc).into())
+}
+
+pub(crate) fn serialize_option_system_time<S>(
```

Comment:
in my head `optional` `serialize_option_system_time`  sounds better but I'll leave it up to you

Reply:
I agree! done